lib/redis.rb

```diff
@@ -2805,11 +2805,13 @@ def method_missing(command, *args)
     }
 
   FloatifyPairs =
-    lambda { |array|
-      if array
-        array.each_slice(2).map do |member, score|
+    lambda { |result|
+      if result.is_a?(Array)
+        result.each_slice(2).map do |member, score|
```

Comment:
We should ask `respond_to?(:each_slice)` rather than specific type checking to allow any kind of Enumerable to work here.